deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/UrlTrie.kt

```diff
@@ -0,0 +1,143 @@
+package com.airbnb.deeplinkdispatch
+
+import com.airbnb.deeplinkdispatch.base.MatchIndex
+import com.airbnb.deeplinkdispatch.base.MatchIndex.*
+import java.io.OutputStream
+import java.nio.charset.Charset
+import kotlin.text.Charsets.UTF_8
+
+data class UriMatch(val uri: DeepLinkUri, val matchId: Int)
+
+@kotlin.ExperimentalUnsignedTypes
+open class TrieNode(open val id: String = "", val type: UByte, open val placeholder: Boolean = false) {
```

Comment:
can a null id be used instead of empty string? generally empty string is a red flag to me in kotlin, but it may makes sense for this implementation